### PR TITLE
Ansible version

### PR DESCRIPTION
**Pull request by patfreeman**
* ansible
  * new repo @ e2r4.com
  * systemd init
  * support new version of ocemr

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,61 +1,38 @@
...
-	
+
 	install --owner=www-data --group=www-data --mode=640 \
 	  config/* $(DESTDIR)/etc/ocemr-appliance
 
-check: $(addprefix promises/,$(POLICY_FILES))
-	
-	( cd promises ; /usr/sbin/cf-promises -Ivf ./promises.cf )
-	
+	if [ ! -d $(DESTDIR)/etc/ocemr-appliance/ansible ]; then \
```
Comment (ph1l) [inline Makefile]: putting all the ansible stuff in /etc/ seems wrong to me, why not `/usr/share/ocemr-appliance` ? We could then create symlinks back to /etc/ocemr-appliance for the actual config files, right?

File: ansible/ocemr.retry [hunk 2]
```diff
@@ -0,0 +1 @@
+localhost
```
Comment (ph1l) [inline ansible/ocemr.retry]: should this be checked in or `.gitignore`'d?

File: ansible/ocemr/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          - ocemr
+      vars:
+          - db_admin_password: supersecretpassword
+          - ocemr_debug: False
+```
+
+Usage
+-----
+
+    # git clone https://github.com/patfreeman/ocemr_ansible.git ocemr
```
Comment (ph1l) [inline ansible/ocemr/README.md]: Is this repo still relevent?

File: ansible/ocemr/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+# handlers file for ocemr
+- name: Initialize MySQL DB
+  shell: i=0;while [ $i -lt 5 ];do if echo "SELECT 1;"|mysql -uroot -p"{{ db_admin_password }}" 2>&1 >/dev/null;then break;else i=$((i+=1));sleep $i;fi;done;PASSWD="{{ db_password }}" MYSQL_ADMIN_PASSWD="{{ db_admin_password }}" /usr/share/ocemr/util/initialize_mysql_db.sh
```
Comment (ph1l) [inline ansible/ocemr/handlers/main.yml:4]: hmm, while this appears to work, I wonder if there would be a cleaner ( and more ansible ) way to do this...

File: ansible/ocemr/tasks/app_config.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+- name: Super User Exists?
+  shell: echo "from django.contrib.auth import get_user_model; User = get_user_model(); print(User.objects.filter(username='{{ ocemr.admin_name }}').count()>0)" | ocemr-manage shell
+  register: superuser_existed
+
+- name: Create Super User
+  shell: ocemr-manage createsuperuser --noinput --username={{ ocemr.admin_name }} --email={{ ocemr.admin_email }}
+  when: superuser_existed.stdout == "False"
+
+- name: Change Super User Password
+  expect:
```
Comment (ph1l) [inline ansible/ocemr/tasks/app_config.yml:23]: what what? expect!! chat scripts, lordy!

neat

File: ansible/ocemr/tasks/packages.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (ph1l) [inline ansible/ocemr/tasks/packages.yml:1]: same question as here: https://github.com/ph1l/ocemr_installer/pull/5#pullrequestreview-168162019

regarding shared infrastructure...

File: ansible/ocemr/tasks/system_config.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+- name: enable serial console
+  systemd:
+    name: serial-getty@ttyS0.service
+    enabled: True
+
+- name: start serial console
+  systemd:
+    name: serial-getty@ttyS0.service
+    state: started
```
Comment (ph1l) [inline ansible/ocemr/tasks/system_config.yml:51]: This task hangs indefinetly in my KVM test environment..

File: ansible/ocemr/templates/ocemr_settings.py.j2 [hunk 8]
```diff
@@ -101,38 +104,45 @@ USE_L10N = True
...
     'django.contrib.admindocs',
+    'django.contrib.staticfiles',
     'ocemr',
 )
 
+ALLOWED_HOSTS = ( '*' )
+
 import version
 OCEMR_VERSION=version.OCEMR_VERSION
-[%CFEngine END %]
```
Comment (ph1l) [inline ansible/ocemr/templates/ocemr_settings.py.j2:138]: CFEngine END, tell me about it

**Review by ph1l — CHANGES_REQUESTED** [review]
In general this looks great and works! Thanks so much for hacking on this whole system.

I do have a couple of nit picks and clarifications...

File: ansible/ocemr/handlers/main.yml (hunk 4 above)
Comment (patfreeman) [reply ansible/ocemr/handlers/main.yml:4]: I've not found one, but am open to suggestions. The issue is that even after mysql has been started by the installation task, it is still not available, thus the sleeping.

File: ansible/ocemr/tasks/system_config.yml (hunk 7 above)
Comment (patfreeman) [reply ansible/ocemr/tasks/system_config.yml]: Might be the serial console part. We can add a variable around it, but I'm unsure how to flip it based on the system installed to.

File: ansible/ocemr/templates/ocemr_settings.py.j2 (hunk 8 above)
Comment (patfreeman) [reply ansible/ocemr/templates/ocemr_settings.py.j2:138]: Yup.

File: Makefile (hunk 1 above)
Comment (patfreeman) [reply Makefile]: I'm fine with the move. Done.

File: ansible/ocemr.retry (hunk 2 above)
Comment (patfreeman) [reply ansible/ocemr.retry]: cruft. cleaned up and ignored.

File: ansible/ocemr/README.md (hunk 3 above)
Comment (patfreeman) [reply ansible/ocemr/README.md]: cruft. cleaned up.

File: ansible/ocemr/tasks/system_config.yml (hunk 7 above)
Comment (patfreeman) [reply ansible/ocemr/tasks/system_config.yml]: Building for an appliance, not a kvm.

File: ansible/ocemr/tasks/packages.yml (hunk 6 above)
Comment (patfreeman) [reply ansible/ocemr/tasks/packages.yml:1]: Let's keep the conversation there.

File: ansible/ocemr/tasks/system_config.yml (hunk 7 above)
Comment (patfreeman) [reply ansible/ocemr/tasks/system_config.yml:51]: Damn chickens! serial-getty must be run after rc.local, but was being called from rc.local. Fixed in the installer: https://github.com/ph1l/ocemr_installer/pull/7/files#diff-8623f738c4ee33ec558f04a28da41399R12

**Review by ph1l — APPROVED** [review]

**Pull request merged**